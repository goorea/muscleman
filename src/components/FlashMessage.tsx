import React from 'react';
import StackFlashMessage from 'react-native-stack-flash-message';
import { useTheme } from '@src/contexts/ThemeProvider';
import Text from '@src/components/Text';
import Icon from '@src/components/Icon';

const FlashMessage: React.FC = () => {
  const { colors } = useTheme();
  const messageWrapperStyle = {
    borderWidth: 1,
    borderColor: colors.greyOutline,
    backgroundColor: colors.background,
  };
  const theme = {
    success: colors.primary,
    info: colors.success,
    error: colors.error,
  };
  const icons = {
    success: (
      <Icon
        name="checkmark-circle-outline"
        size={20}
        color="primary"
        type="ionicon"
      />
    ),
    info: (
      <Icon
        name="information-circle-outline"
        size={20}
        color="success"
        type="ionicon"
      />
    ),
    error: (
      <Icon
        name="close-circle-outline"
        size={20}
        color="error"
        type="ionicon"
      />
    ),
  };

  return (
    <StackFlashMessage
      ref={ref => StackFlashMessage.setRef(ref)}
      messageWrapperStyle={messageWrapperStyle}
      titleComponent={Text}
      contentsComponent={Text}
      theme={theme}
      icons={icons}
    />
  );
};

export default FlashMessage;
