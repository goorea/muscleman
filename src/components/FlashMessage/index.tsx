import React from 'react';
import StackFlashMessage from 'react-native-stack-flash-message';
import { useTheme } from '@src/contexts/ThemeProvider';
import Text from '@src/components/Text';
import useFlashMessageProps from './hooks/useFlashMessageProps';

const FlashMessage: React.FC = () => {
  const { colors } = useTheme();
  const { messageWrapperStyle, theme, icons } = useFlashMessageProps(colors);

  return (
    <StackFlashMessage
      ref={ref => StackFlashMessage.setRef(ref)}
      messageWrapperStyle={messageWrapperStyle}
      titleComponent={Text}
      contentsComponent={Text}
      theme={theme}
      icons={icons}
    />
  );
};

export default FlashMessage;
