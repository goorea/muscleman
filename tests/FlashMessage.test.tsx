import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import FlashMessage from '@src/components/FlashMessage';

describe('FlashMessage 컴포넌트', () => {
  it('렌더링이 올바르게 된다', () => {
    renderer.create(<FlashMessage />);
  });
});
