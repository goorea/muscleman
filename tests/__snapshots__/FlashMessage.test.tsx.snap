// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashMessage 컴포넌트 렌더링이 올바르게 된다 1`] = `
<View
  style={
    Array [
      Object {
        "left": "5%",
        "marginVertical": -5,
        "position": "absolute",
        "right": "5%",
        "top": 20,
        "zIndex": 1050,
      },
      undefined,
    ]
  }
/>
`;
